StripeApplePay/StripeApplePay/Source/ApplePayContext/STPApplePayContext.swift

```diff
@@ -573,6 +595,13 @@ public class STPApplePayContext: NSObject, PKPaymentAuthorizationControllerDeleg
                 guard let clientSecret = clientSecret, intentCreationError == nil,
                     self.authorizationController != nil
                 else {
+                    let errorMessage = "Failed on payment method completion"
+                    let errorAnalytic = ErrorAnalytic(event: .unexpectedApplePayError,
+                                                      error: InternalError.invalidState,
+                                                      additionalNonPIIParams: [
+                                                        "error_message": errorMessage,
+                                                      ])
+                    self.analyticsClient.log(analytic: errorAnalytic, apiClient: self.apiClient)
                     handleFinalState(.error, intentCreationError)
```

Comment:
What is handleFinalState? Can we make the analytics calls in there to consolidate?

Reply:
Thanks for the feedback!  I was trying to get unique messages populated in the handleFinalState, but passing only the analytic seems to make the code cleaner.